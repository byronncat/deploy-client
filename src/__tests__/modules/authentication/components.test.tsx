import { BrowserRouter } from 'react-router-dom';
import { act, fireEvent, render } from '@testing-library/react';
import {
  Brand,
  Form,
  FIELD,
  DEFAULT_VALUES,
  Divider,
  NavigationText,
} from '@authentication';

describe('components', () => {
  it('should render brand component correctly', () => {
    const { container } = render(<Brand />);
    expect(container).toMatchSnapshot();
  });

  describe('form', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Form
          fieldList={FIELD.LOGIN}
          defaultValues={DEFAULT_VALUES.LOGIN_FORM}
          submitHandler={() => console.log('submit')}
          submitPlaceholder="Login"
        />,
      );
      expect(container).toMatchSnapshot();
    });

    it('should call submitHandler when the form is submitted', async () => {
      const submitHandler = jest.fn();
      const { container } = render(
        <Form
          fieldList={FIELD.LOGIN}
          defaultValues={DEFAULT_VALUES.LOGIN_FORM}
          submitHandler={submitHandler}
          submitPlaceholder="Login"
        />,
      );
      await act(() => {
        const form = container.querySelector('form');
        if (form) fireEvent.submit(form);
      });
      expect(submitHandler).toHaveBeenCalledTimes(1);
    });
  });

  it('should render divider component correctly', () => {
    const { container } = render(<Divider />);
    expect(container).toMatchSnapshot();
  });

  it('should render navigation component correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <NavigationText
          navigateText="new user?"
          text="register"
          path="/register"
        />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
