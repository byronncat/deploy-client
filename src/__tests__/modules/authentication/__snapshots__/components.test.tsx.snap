// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components form should render correctly 1`] = `
<div>
  <form>
    <div>
      <input
        class=""
        id="email"
        name="email"
        placeholder=" "
        type="email"
      />
      <label
        class=""
        for="email"
      >
        Email
      </label>
    </div>
    <p
      class=""
    />
    <div>
      <input
        class=""
        id="password"
        name="password"
        placeholder=" "
        type="password"
      />
      <label
        class=""
        for="password"
      >
        Password
      </label>
      <span
        class="flex justify-center items-center absolute top-0 right-0 h-full w-12 text-xl cursor-pointer transition duration-300 opacity-50 hover:opacity-100"
      >
        <i
          class="icon-eye"
        />
      </span>
    </div>
    <p
      class=""
    />
    <input
      class="button w-full mt-5"
      type="submit"
      value="Login"
    />
  </form>
</div>
`;

exports[`components should render brand component correctly 1`] = `
<div>
  <img
    alt="logo"
    class="block"
    height="60"
    src="logo.svg"
    width="60"
  />
  <h1
    class="text-neon-glowing-1 mt-2 mb-6 font-monoton text-3xl tracking-widest -skew-y-6"
  >
    bygram
  </h1>
</div>
`;

exports[`components should render divider component correctly 1`] = `
<div>
  <p
    class="text-center my-2"
  >
    --- or ---
  </p>
</div>
`;

exports[`components should render navigation button component correctly 1`] = `
<div>
  <a
    class="block font-medium text-center capitalize duration-300 hover:text-slate-300"
    href="/register"
  >
    register
  </a>
</div>
`;
